onnx_tf/backend.py

```diff
@@ -267,43 +267,57 @@ def supports_device(cls, device):
 
   @classmethod
   def onnx_graph_to_tensorflow_ops(cls,
-                                   graph_def,
+                                   subgraph,
                                    input_values,
+                                   tensor_dict,
                                    opset=None,
                                    strict=True):
     """
     Converts ONNX graph to Tensorflow operations
     Args:
-      graph_def:        the ONNX graph to be converted
+      subgraph:         the ONNX graph to be converted
       input_values:     dictionary with values/tensors to initialize
-                        the graph inputs. the dictionary must contain values
-                        for all the graph_def.input
+                        the subgraph inputs. if the subgraph.input
+                        are send in as parameters then it is required,
+                        otherwise this can be empty dictionary
+      tensor_dict:      the dictionary that contain values for all the
+                        node.inputs in the subgraph that are not defined
+                        in the subgraph or input_values.
       opset:            opset version of the operator set.
       strict:           whether to enforce semantic equivalence between the
                         original model and the converted tensorflow model,
                         defaults to True (yes, enforce semantic equivalence).
     Returns:
       array of Tensorflow Tensors
     """
-    input_dict_items = []
-    # set input values for the subgraph
-    for value_info in graph_def.input:
-      if value_info.name in input_values:
-        x = input_values[value_info.name]
-        input_dict_items.append((value_info.name, x))
-
-    tensor_dict = dict(input_dict_items)
-
-    for node in graph_def.node:
+    # get the subgraph.input from input_values
+    subgraph_tensor_dict = dict(input_values)
+    # get the rest of the subgraph input from tensor_dict
+    for i in subgraph.input:
+      if i.name not in subgraph_tensor_dict.keys():
+        subgraph_tensor_dict[i.name] = tensor_dict[i.name]
+    # get the required initializer constant node(s) for the subgraph
+    # Need to get the initializer constant nodes from tensor_dict here
+    # because input from initializer will not be send in as inputs
+    # to the subgraph and those nodes are not in the subgraph
+    nodes_outputs = []
+    for node in subgraph.node:
+      for o_name in node.output:
+        nodes_outputs.append(o_name)
+    for node in subgraph.node:
+      for i_name in node.input:
+        if i_name not in nodes_outputs and i_name not in subgraph_tensor_dict.keys():
+          subgraph_tensor_dict[i_name] = tensor_dict[i_name]
+
+    for node in subgraph.node:
```

Comment:
This line for... doesn't seem needed. As long as we have the input values for a node, we should be able to convert it to TF ops.

Reply:
@chinhuang007 updated, please review again.